### DISCUSSION TITLE
showfnotes not re-ordering counter on transclusion

### DISCUSSION THREAD

**Discussion by dhjackal** [kookma/Refnotes#41] [General]
Hi. 

There appears to be a bug with re-ordering the fnote counter on tiddler transclusion. I.e. the counter continues for the actual note BUT restarts in the showfnotes call at the bottom of the tiddler. For example

tiddler 1 contains :- 

This is a test <<fnote "test">>

showfnotes

tiddler 2 contains -  

This is another test <<fnote "another test">>

showfnotes 

tiddler 3 transcludes tiddler 1 & 2 

{{tiddler 1}}
{{tiddler 2}}

results are 

This is a test^1

    1.test

This is another test^2

   1.another test

Should the counter reset after each tiddler?? for consistency? 

Thanks for the plugin btw.......excellent stuff. If i can help or contribute in any way please let me know. 

dhjackal

**Comment by kookma** [comment]
I think the way you have used makes to produce this issue

When a tiddler transclude several tiddlers, we again call showfnotes there! See example: https://kookma.github.io/TW-Refnotes/#demo%2Ffootnotes-transclusion

It does not make sense to have footnote in the middle of tiddler for several time!

**Comment by kookma** [comment]
> Thanks for the plugin btw.......excellent stuff. If i can help or contribute in any way please let me know.

Many thanks, I welcome such collaboration! See the Ideas and Todo section.

**Comment by dhjackal** [comment]
Ah. I see and now understand. Apologies it was a mistake on my behalf expecting something that shouldn't be. I'm fairly new to tiddlywiki but love the flexibility and adaptablity and the limitless scope for my personal purposes. So far I have managed to get it doing pretty much everything I require but have greatfully found and used the kookma plugins for a host of ad-hoc scenarios and solutions. Many many thanks.

**Comment by dhjackal** [comment]
Actually Mohammad, 

How difficult (or is it possible) to achive this? So, my working scenario is as follows......I have a book containing several pages (individual tiddlers).....each page contains it's own set of page/footnotes. I then want to grab a sub-section of the pages say for instance the first page of every chapter and transclude them together....each "page" displaying it's specific pagenotes.....I don't want to rollup the pagenotes at the end of the transcluded list rather render the pages as it actually is including footnotes for that page. Perhaps a switch, variable to "turn on/off" the flow of the fnote macro over transcluded tiddlers??? Apologies if this doesn't make sense. It is probably the way i want it to work v the way it does and should work. Just wondering. Thanks again.

**Comment by kookma** [comment]
You can use a viewtemplate to NOT include hardcoded `<<showrefnotes>>` in any of those thiddler. See example https://kookma.github.io/TW-Refnotes/#demo%2Ffootnote%2Fshow%20footnotes%20automatically

Then if you transclude such tiddlers (page), then the content will be transcluded not the foornotes. 
You can in the master tiddler (the one transclude other pages) include `<<showrefnotes>>` again to show all footnotes from those tiddlers.

By the way I highly recommend to join [talk.tiddlywiki.org](talk.tiddlywiki.org), you will get more intresting answers!

**Comment by dhjackal** [comment]
Thanks Mohammad,

I understand now.

And I appreciate your suggestion to join [talk.tiddlywiki.org]talk.tiddlywiki.org. I will do that.

Thanks again,

Declan
________________________________

**Comment by dhjackal** [comment]
Mohammad,

Another quick enquiry. Would you like me / contributors to suggest edits for your instructions? Fix Ggammatical errors or typos etc? If you want I will suggest updates / changes as I read through the documentation.

If this helps please let me know and let me know the best way to make or suggest edits.

Thanks,

Declan
________________________________

**Reply by kookma** [reply]
> Another quick enquiry. Would you like me / contributors to suggest edits for your instructions? Fix Ggammatical errors or typos etc? If you want I will suggest updates / changes as I read through the documentation.

I absolutely such collaboration and I am sure there is many places in docs to be corrected for English! So please go ahead!
You can edit tiddlers and then export them and share here as Zip file, or you can push PR which one is easier for you!

Thank you

**Comment by dhjackal** [comment]
Another note. Having read the instructions (carefully) and followed your prompt I have got the desired outcome for the scenario I was facing SO I just wanted to say thank you again.

Declan
________________________________

**Comment by dhjackal** [comment]
Mohammad, 

Now i fully see the problem. It's not the showfnotes macro but fnote. So, what is happening is the first tiddler page is rendered correctly with the correct fnotes. The second continues the numbering from the fnote in the first tiddler (probably expected behaviour, under normal conditions)......BUT, what i want to do is transclude multiple pages including their fnotes and footnotes as they are rendered individually. I am trying to render the pages as they look, no pull through or continuation of fnote or showfnotes.......just the page and it's notes......followed by another page and it's notes but it's being clever and trying to rebuild the page and the notes for the transcluded tiddlers. I am probably doing something wrong???

**Reply by kookma** [reply]
> probably expected behaviour

Yes, this is the expected behavior, and it is applied from this CSS rule: https://kookma.github.io/TW-Refnotes/#%24%3A%2Fplugins%2Fkookma%2Frefnotes%2Fstyles%2Ffootnote-counter

At least whit current implementation I cannot see a solution!

**Comment by dhjackal** [comment]
That is fine. I understand and I can use the macros as they are and it will work for me. I just thought I would ask. No harm in asking. Plus my understanding of github, TW-Refnotes and tiddlywiki has increased so this has all been benefical. 

Every day is a learning day!!

Thanks again Mohammad.
